ArduCopter/mode_rtl.cpp

```diff
@@ -450,11 +450,20 @@ void ModeRTL::compute_return_target()
     rtl_path.return_target = ahrs.get_home();
 #endif
 
-    // curr_alt is current altitude above home or above terrain depending upon use_terrain
+    // curr_alt is current altitude which can be above home, above terrain or absolute
     int32_t curr_alt = copter.current_loc.alt;
-
     // determine altitude type of return journey (alt-above-home, alt-above-terrain using range finder or alt-above-terrain using terrain database)
     ReturnTargetAltType alt_type = ReturnTargetAltType::RELATIVE;
+
+    if (get_alt_type() == RTLAltType::RTL_ALTTYPE_ABSOLUTE) {
+        alt_type = ReturnTargetAltType::ABSOLUTE;
+        if (!copter.current_loc.get_alt_cm(Location::AltFrame::ABSOLUTE, curr_alt)) {
+            // just in case we fail to get current altitude in ABSOLUTE frame
+            curr_alt = 0;
+            gcs().send_text(MAV_SEVERITY_WARNING, "RTL: unexpected error calculating current alt-above-sea");
```

Comment:
Shouldn't we fall back to doing stuff relatively if we fail to get ab absolute height?  The other cases in here seem to fall back to relative.

Reply:
Yes. Fixed it. Thanks.